config/routes.rb

```diff
@@ -19,6 +19,7 @@
   resources :nodes, only: [:index, :show, :destroy] do
     collection do
       post :bootstrap
+      post :assign_roles
```

Comment:
good question, i'd maybe discuss that first and do a follow up, as that requires some architectural planning like what should the roles controller do beyond that etc.